main.go

```diff
@@ -23,76 +23,162 @@ func readFileLines(path string) ([]string, error) {
+ *
+ * Each individual line may fail to parse if the file is malformed,
+ * and this error is passed along per line in the channel returned
+ *
+ * This channel is safe to read until completion
+ */
+func readPhrasesFromFile(filename string) chan PhraseReadResults {
+	phraseResults := make(chan PhraseReadResults)
+
+	go func() {
+		defer close(phraseResults)
+
+		phrases, err := readFileLines(filename)
 		if err != nil {
 			phraseResults <- PhraseReadResults{nil, err}
 			return
 		}
-		parsedPhrases = append(parsedPhrases, parsedPhrase)
+
+		for _, phrase := range phrases {
+			parsedPhrase, err := datatypes.ParsePhrase(phrase)
+			if err != nil {
+				phraseResults <- PhraseReadResults{nil, err}
+				return
+			} else {
+				phraseResults <- PhraseReadResults{parsedPhrase, nil}
+			}
+		}
+	}()
+
+	return phraseResults
+}
+
+type PhraseWav struct {
+	phrase            *datatypes.Phrase
+	phraseWavFilePath string
+}
+
+type PhraseWavResults struct {
+	phraseWav *PhraseWav
+	err       error
+}
+
+/**
+ * Management function for spawning go routines to generate WAV files and manage the channel to do so
+ *
+ * The channel is safe to read until completion
+ */
+func generateWavFilesForPhrases(tpv *datatypes.TaskProcessVariables, phraseResults chan PhraseReadResults) chan PhraseWavResults {
+	phrasesGenerated := make(chan PhraseWavResults)
+
+	go func() {
+		defer close(phrasesGenerated)
+
+		var wg sync.WaitGroup
+
+		for phraseResult := range phraseResults {
+			if phraseResult.err != nil {
+				phrasesGenerated <- PhraseWavResults{nil, phraseResult.err}
+				return
+			}
+
+			phrase := phraseResult.phrase
+
+			wg.Add(1)
+			go func() {
+				defer wg.Done()
+				generateWavFileForPhrase(tpv, phrase, phrasesGenerated)
+			}()
+		}
+
+		wg.Wait()
+	}()
+
+	return phrasesGenerated
+}
+
+func generateWavFileForPhrase(tpv *datatypes.TaskProcessVariables, phrase *datatypes.Phrase, phrasesGenerated chan PhraseWavResults) {
+	err := (*tpv.Generator).GenerateAudioFile(tpv.Parameters, phrase.PhraseText, tpv.GetPhraseFilePath(phrase.PhraseIndex))
+
+	if err != nil {
+		phrasesGenerated <- PhraseWavResults{nil, err}
+		return
 	}
 
-	phraseResults <- PhraseReadResults{parsedPhrases, nil}
+	phrasesGenerated <- PhraseWavResults{&PhraseWav{phrase, tpv.GetPhraseFilePath(phrase.PhraseIndex)}, nil}
 }
 
+/**
+ * Process task pipeline
+ *
+ * Goes through initializing and managing each step:
+ * - Convert input audio to WAV
+ * - Parse provided phrases
+ * - Generate WAV files from parsed phrases
+ * - Prepare for MFCC
+ * - Prepare for DTW
+ */
 func processTask(results chan string, task *datatypes.Task, generator *datatypes.AudioGenerator, tempDir string) {
 	tpv := datatypes.NewTaskProcessVariables(task, generator, tempDir)
 	defer func() {
 		results <- tpv.GetFinalLogs()
 	}()
 
 	if len(task.Description) > 0 {
-		tpv.Println("")
-		tpv.Println("*** ", tpv.Task.Description, " ***")
-		tpv.Println("")
+		fmt.Println("")
```

Comment:
Why did you change this from `tpv.Println()` which includes it in the logs for the task?

Reply:
That was for debugging. I'll redo the commit to go back to tpv.Println